File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/AssignmentsTree/Resource.cs
using DlhSoft.Windows.Controls;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Demos.WPF.CSharp.GanttChartDataGrid.AssignmentsTree
{
    public class Resource : DataTreeGridItem
    {
        private double allocation;
        public double Allocation { get { return allocation; } set { allocation = value; OnPropertyChanged(nameof(Allocation)); } }

        private string role;
        public string Role { get { return role; } set { role = value; } }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples/WPF-VisualBasic/GanttChartDataGrid/AssignmentsTree/Resource.vb
Imports DlhSoft.Windows.Controls
Imports System.Text

Public Class Resource
    Inherits DataTreeGridItem

    Private allocationValue As Double
    Public Property Allocation() As Double
        Get
            Return allocationValue
        End Get
        Set(value As Double)
            allocationValue = value
            OnPropertyChanged(NameOf(Allocation))
        End Set
    End Property

    'INSTANT VB NOTE: The variable role was renamed since Visual Basic does not allow variables and other class members to have the same name:
    Private roleValue As String
    Public Property Role() As String
        Get
            Return roleValue
        End Get
        Set(value As String)
            roleValue = value
        End Set
    End Property
End Class
